### PR TITLE
Sprint 10

### PR DESCRIPTION
**Pull request by codyseibert**
### Bugs: Found during Sprint 10 Review Prep #1140

- various bugs to fix

### Petitions Clerk: Review Party Type, Contact Information, and other Metadata #699

- redesign to put into different tabs

Screen Shot 2019-03-19 at 11 36 26 AM

### Petitions Clerk: Orders to be Issued following QC process #698

- adding various check boxes to the QC document details edit page which allows the petitionsclerk to mark orders

Screen Shot 2019-03-19 at 11 35 49 AM

### Create an AWS diagram #785

- created an aws diagram and added it to the repo (this diagram was created by hand)

### Bug: 401 errors not sending users back to login screen #966

- fixing a bug which occurred when the user's token was expired but the app wasn't redirecting the user back to cognito to log in

### Court User: Start a message thread #962

- ability to create a new message ( work item) on a document

Screen Shot 2019-03-19 at 11 33 57 AM

Screen Shot 2019-03-19 at 11 34 06 AM

### Petitioner: Statement of Taxpayer Identification Number #786

- requiring the STIN document when creating a petition

Screen Shot 2019-03-19 at 11 32 51 AM

### Document-level message thread in document detail #359

- added sub tabs on the messages tab on the document details which allows a user to view all the pending / completed work items

Screen Shot 2019-03-19 at 11 31 20 AM

### Court: Ability to correct a case caption #310 

- Added an "edit" button next to the case caption which displays a edit modal.

Screen Shot 2019-03-19 at 11 29 11 AM

Screen Shot 2019-03-19 at 11 29 34 AM

## Deploy Steps
- You will need to run the management/management/bin/setup-jobs.sh script for these changes to deploy because we separated the deploy steps into separate jobs.  Please refer to step 11 in the SETUP.md file

### CODE REVIEW

**Comment by waldoj** [comment]
I'm going to try something different with this pull request — I'm going to include the code review checklist here, and update it as I proceed. That'll make it easier for me to track my progress, as well as for others to track my progress.

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, and running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: build-all-local.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash -e
```
[the remark this answers is not in the document]
Comment (codyseibert) [reply build-all-local.sh]: we could probably just make them all `#!/usr/bin/env bash`... I'm not too which one is "correct"

[the remark this answers is not in the document]
Comment (waldoj) [reply build-all-local.sh]: It's my (shallowly-rooted) understanding that `#!/usr/bin/env bash` is "correct" — as in, like, I once read it on StackOverflow or something.

[the remark this answers is not in the document]
Comment (codyseibert) [reply build-all-local.sh]: I made them all consistent

**Pull request merged**